tuf/api/metadata.py

```diff
@@ -1153,17 +1154,21 @@ def from_dict(cls, delegations_dict: Dict[str, Any]) -> "Delegations":
         for keyid, key_dict in keys.items():
             keys_res[keyid] = Key.from_dict(keyid, key_dict)
         roles = delegations_dict.pop("roles")
-        roles_res = []
+        roles_res: "OrderedDict[str, DelegatedRole]" = OrderedDict()
         for role_dict in roles:
             new_role = DelegatedRole.from_dict(role_dict)
-            roles_res.append(new_role)
+            if roles_res.get(new_role.name) is not None:
+                raise exceptions.RoleAlreadyExistsError(
+                    f"Delegated role names must be unique! Got {new_role.name}"
+                )
```

Comment:
This is code that gets called from serialization (which wraps it in a catch all try-except) so using custom exceptions seems not useful.

repository code that wants to add a roles is now going to use  `targets.delegation.roles[role.name] = role` so the exception won't be useful in that case either, as far as I can see...

Reply:
Okay, will change the exception to `ValueError`  and will shorten the message.